LoopUI/Charts/PredictedGlucoseChart.swift

```diff
@@ -107,7 +107,7 @@ extension PredictedGlucoseChart {
 
                 if displayedScheduleOverride != nil {
                     if displayedScheduleOverride!.scheduledEndDate > preMealOverride.scheduledEndDate {
-                        displayedScheduleOverride!.scheduledInterval = DateInterval(start: preMealOverride.scheduledEndDate, end: displayedScheduleOverride!.scheduledEndDate)
+                        displayedScheduleOverride!.scheduledInterval = DateInterval(start: displayedScheduleOverride!.startDate, end: displayedScheduleOverride!.scheduledEndDate)
```

Comment:
Thanks for the PR. I think this makes sense for when the scheduled override starts after premeal ends, but what about the case where the scheduled override starts *before* premeal ends?  I think this will draw a section of the timeline that shows two target ranges, which would be confusing. Perhaps the max of (premeal.end and scheduled.start) should be used here.